### PR TITLE
EKF2: Robustness Improvements

### PR DESCRIPTION
**Pull request by priseborough**
Retuned to reduce sensitivity to GPS errors: 
http://diydrones.com/forum/topics/pixracer-with-ardupilot?Adds detection, isolation and reporting of fusion numerical errors that would destabilise the covariance matrix and may be the cause of a reported loss of attitude reference after takeoff.
http://discuss.ardupilot.org/t/ekf2-errors-at-startup/8872
Has been tuned on the following logs using the replay program with the --no-params option set (makes replay use the default parameters) : 

http://uav.tridgell.net/ReplayLogs/

Copter compatibility has been checked in SITL with auto mission and in loiter with continuous yaw spins.

Further copter replay logs from flight testing containing high rate and high 'g' flight manoeuvres are required to complete tuning and validation. Please see the following for information on how to log the required data: 

http://discuss.ardupilot.org/t/new-log-disarmed-and-log-replay-parameters/8926

### CODE REVIEW

**Comment by priseborough** [comment]
SITL takeoffs with all COMPASS_USE parameters set to 0 are failing. Further work required.

File: libraries/AP_NavEKF/AP_NavEKF_core.cpp [hunk 1]
```diff
@@ -3918,6 +3918,11 @@ void NavEKF_core::readIMUData()
         // just read primary gyro
         readDeltaAngle(ins.get_primary_gyro(), dAngIMU);
     }
+    //debug testing
+    if (imuSampleTime_ms > 300E3) {
+        dAngIMU.x += 0.01745f * dtDelAng;
+    }
+
```
Comment (OXINARF) [inline libraries/AP_NavEKF/AP_NavEKF_core.cpp]: Is this _debug testing_ to stay here?

**Comment by priseborough** [comment]
Testing on plane the logs from http://uav.tridgell.net/ReplayLogs/ with default parameters and magnetometer disabled has been successfully completed. Th efollowing are the summary normalised innovation plots from each flight.

eaglet-2016-05-14-2
eaglet-2016-05-14
eaglet-2016-05-17-leonard
ranger-2016-05-07-2

**Comment by tridge** [comment]
thanks for all your work on this Paul! I've pushed it to master

**Comment by cudnn** [comment]
hi may i use quadplane control x plusone drone？ https://m.youtube.com/watch?v=pOG7lF0tYiM

**Pull request closed without merging**